### PR TITLE
Fix code adding websocket headers

### PR DESCRIPTION
**Pull request by cvb941**
As discussed in #4123

### CODE REVIEW

File: docs/source/advanced/authentication.mdx [hunk 1]
```diff
@@ -52,7 +52,7 @@ val apolloClient = ApolloClient.Builder()
     .wsProtocol(
         SubscriptionWsProtocol.Factory(
             connectionPayload = {
-              mapOf("authorization" to token)  // highlight-line
+              mapOf("headers" to mapOf("authorization" to token))   // highlight-line
```
[the remark this answers is not in the document]
Comment (cvb941) [reply docs/source/advanced/authentication.mdx:55]: Are we allowed to mention Hasura in the docs?

[the remark this answers is not in the document]
Comment (BoD) [reply docs/source/advanced/authentication.mdx:55]: I think that would be a bit out of place in this general doc. Also in my tests this way of doing it didn't seem to apply to all Hasura servers (didn't work on the [Todo sample](https://hasura.io/learn/graphql/graphiql))

[the remark this answers is not in the document]
Comment (martinbonnin) [reply docs/source/advanced/authentication.mdx:55]: Maybe add both? 

```
            connectionPayload = {
              // some servers use "authorization" directly
              mapOf("authorization" to token)   // highlight-line
              // others require to wrap in a "headers" map
              mapOf("headers" to mapOf("authorization" to token))   
              // others will expect different maps
              // refer to your backend docs for the actual map to use
            }
```

**Review by BoD — APPROVED** [review]

**Comment by martinbonnin** [comment]
@cvb941 I went ahead and pushed the code for both solutions (and a comment for potentially other solutions). I hope you don't mind. Let me know otherwise and I can rework. Thanks for investigating this in all cases 😊 !

**Pull request merged**